app/javascript/packs/components/AddNewRestaurant.jsx

```diff
@@ -0,0 +1,176 @@
+import React, { Component } from 'react';
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+import MenuItem from 'material-ui/MenuItem';
+import Slider from 'material-ui/Slider';
+import Checkbox from 'material-ui/Checkbox';
+import { TextValidator, ValidatorForm, SelectValidator } from 'react-material-ui-form-validator';
+import PropTypes from 'prop-types';
+import { addRestaurant } from '../utils/api';
+
+class AddNewRestaurant extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      openDialog: false,
+      formData: {
+        name: '',
+        address: '',
+        cuisine: '',
+        max_delivery_time: 120,
+        accepts_10bis: false,
+      },
+    };
+
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleTextChange = this.handleTextChange.bind(this);
+    this.handleSelectChange = this.handleSelectChange.bind(this);
+    this.handleSliderChange = this.handleSliderChange.bind(this);
+    this.handleCheckboxChange = this.handleCheckboxChange.bind(this);
+    this.clearState = this.clearState.bind(this);
+  }
+  clearState() {
+    const { formData } = this.state;
+    formData.name = '';
+    formData.address = '';
+    formData.cuisine = '';
+    formData.max_delivery_time = 120;
+    formData.accepts_10bis = false;
+    this.setState(formData);
+  }
+  handleOpen() {
+    this.setState({ openDialog: true });
+  }
+
+  handleClose() {
+    this.setState({ openDialog: false });
+    this.clearState();
+  }
+
+  handleSubmit() {
+    addRestaurant(this.state.formData)
+      .then(() => {
+        this.props.updateRests();
+        this.handleClose();
+        this.clearState();
+      })
+      .catch(err => console.log(err));
+  }
+
+  handleTextChange(event) {
+    const { formData } = this.state;
+    formData[event.target.name] = event.target.value;
+    this.setState({ formData });
+  }
+
+  handleCheckboxChange() {
+    const formData = this.state.formData;
+    formData.accepts_10bis = !formData.accepts_10bis;
+    this.setState({ formData: formData });
+  }
+
+  handleSelectChange(event, index, value) {
+    const formData = this.state.formData;
+    formData.cuisine = value;
+    this.setState({ formData: formData });
+  }
+
+  handleSliderChange(event, value) {
+    const formData = this.state.formData;
+    formData.max_delivery_time = value;
+    this.setState({ formData: formData });
+  }
+
+  render() {
+    const cuisines = this.props.cuisines;
```

Comment:
you can use here the `{...} = this.props` syntax as well

Reply:
👍